xbmc/guilib/GUIWindowManager.cpp

```diff
@@ -771,9 +771,16 @@ void CGUIWindowManager::ActivateWindow_Internal(int iWindowID, const std::vector
   // debug
   CLog::Log(LOGDEBUG, "Activating window ID: %i", iWindowID);
 
+  // make sure we check mediasources too
+  if (params.size() > 0)
+    g_passwordManager.strMediasourcePath = params[0];
```

Comment:
Just looking at strMediaSourcePath, once it gets set here, it'll stay set. Do we need to set it to empty in CheckMenuLock after it's been processed?

Reply:
done, now i built and a quick check looks good to me. let me know if somethings still missing